ton-smartcontract/src/commonMain/kotlin/org/ton/smartcontract/ChunkedData.kt

```diff
@@ -0,0 +1,35 @@
+package org.ton.smartcontract
+
+import org.ton.cell.Cell
+import org.ton.cell.CellBuilder
+import org.ton.cell.CellSlice
+import org.ton.hashmap.HashMapE
+import org.ton.tlb.*
+import org.ton.tlb.constructor.tlbCodec
+
+data class ChunkedData(
+    val data: HashMapE<SnakeDataTail>
+) {
+    companion object : TlbCodec<ChunkedData> by ChunkedDataCombinator {
+        @JvmStatic
+        fun tlbCodec(): TlbCombinator<ChunkedData> = ChunkedDataCombinator
+    }
+}
+
+private object ChunkedDataCombinator : TlbCombinator<ChunkedData>() {
+    override val constructors: List<TlbConstructor<out ChunkedData>> by lazy { listOf(ChunkedDataConstructor) }
+
+    override fun getConstructor(value: ChunkedData): TlbConstructor<out ChunkedData> = ChunkedDataConstructor
+
+    private object ChunkedDataConstructor : TlbConstructor<ChunkedData>(
+        schema = "chunked_data#_ data:(HashMapE 32 ^(SnakeData ~0)) = ChunkedData;"
+    ) {
+        private val dataCodec = HashMapE.tlbCodec(32, Cell.tlbCodec(SnakeDataTail.tlbCodec()))
```

Comment:
`SnakeDataTail`? Maybe `SnakeData`?

Reply:
No, exactly SnakeDataTail since it is defined as `(Snake Data ~0)`